scripts/create_pelorus_operator.sh

```diff
@@ -136,19 +151,43 @@ pushd "${destination_dir}" || exit 2
     rm helm-charts/pelorus/charts/*.tgz
 
     # Correct operator version, to be latest +1
-    OPERATOR_VERSIONS=$(curl -H "Authorization: Bearer XYZ" -X GET "https://quay.io/api/v1/repository/${OPERATOR_ORG_NAME}/${OPERATOR_PROJECT_NAME}/tag/" | jq ".tags[] | select(.end_ts == null) | .name" | sed -e 's|\"||g' | grep -E '^[0-9]+\.[0-9]+\.[0-9]+$' -o | sort -t. -nrk1,1 -nrk2,2 -nrk3,3)
+    OPERATOR_VERSIONS=$(curl -H "Authorization: Bearer XYZ" -X GET "https://quay.io/api/v1/repository/${OPERATOR_ORG_NAME}/${OPERATOR_PROJECT_NAME}/tag/" 2> /dev/null | jq ".tags[] | select(.end_ts == null) | .name" | sed -e 's|\"||g' | grep -E '^[0-9]+\.[0-9]+\.[0-9]+$' -o | sort -t. -nrk1,1 -nrk2,2 -nrk3,3)
     OPERATOR_VERSION=$(echo "$OPERATOR_VERSIONS" | head -1 )
     echo "INFO: Current operator version: ${OPERATOR_VERSION}"
 
-    if [ -z ${operator_ver+x} ]; then
-        NEW_OPERATOR_VERSION=$(echo "${OPERATOR_VERSION}" | awk -F. -v OFS=. '{$NF += 1 ; print}')
-        # Case where operator was never in the repo
-        if [ "$NEW_OPERATOR_VERSION" -eq "1" ]; then
-          NEW_OPERATOR_VERSION=0.0.1
+    OPERATOR_MAKEFILE_IN_MASTER="$(curl https://raw.githubusercontent.com/dora-metrics/pelorus/master/pelorus-operator/Makefile 2> /dev/null)"
+    OPERATOR_VERSION_IN_MASTER=$(echo "$OPERATOR_MAKEFILE_IN_MASTER" | grep "VERSION ?= " | cut -c 12-)
+    # shellcheck disable=SC2207
+    CURRENT_VERSION_ARRAY=( $(echo "$OPERATOR_VERSION_IN_MASTER" | sed "s/-rc//g" | tr ' . '  '  ') )
+    V_X_VER=${CURRENT_VERSION_ARRAY[0]}
+    V_Y_VER=${CURRENT_VERSION_ARRAY[1]}
+    V_Z_VER=${CURRENT_VERSION_ARRAY[2]}
+    V_RC_VER=${CURRENT_VERSION_ARRAY[3]}
+
+    RC_SUFFIX=""
```

Comment:
I think bash is not good for this kind of thing

I can merge them on this PR to avoid duplication, or wait to do what I want to do, that is rewrite all complex bash scripts in Python or other language, in another task

Reply:
Ok that's fine. I will be ok to merging with duplication, but how about small python helper script to include such function ?

Reply:
merged the 3 scripts in a python one